felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -2031,6 +2083,39 @@ func (m *bpfEndpointManager) GetRouteTableSyncers() []routetable.RouteTableSynce
 	return tables
 }
 
+// updatePolicyCache modifies entries in the cache, adding new entries and marking old entries dirty.
+func (m *bpfEndpointManager) updatePolicyCache(name string, owner string, inboundRules, outboundRules []*proto.Rule) {
+	var ruleIds set.Set[bpfRuleInfo]
+	if val, ok := m.polNameToRuleIDs[name]; ok {
+		// If the policy name exists, it means the policy is updated. There are cases where both inbound,
+		// outbound rules are updated or any one.
+		// Mark all the entries as dirty.
+		ruleIds = markAllRuleIdsDirty(val)
```

Comment:
this could be a dirty.AddAll(val.Slice()) or can be wrapped in a .Join() operation in the set lib now that we have it all typed.